pkg/reconciler/v1alpha1/pipelinerun/pipelinerun.go

```diff
@@ -220,6 +222,14 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1alpha1.PipelineRun) er
 		return nil
 	}
 
+	var defaults []v1alpha1.PipelineParam
+	if len(p.Spec.Params) > 0 {
+		defaults = append(defaults, p.Spec.Params...)
```

Comment:
why not append contents of `p.Spec.Params` to `ApplyParameters` function as parameters? 

```
p = resources.ApplyParameters(p, pr, p.Spec.Params...)
```

Reply:
No good reason - I was just copying from how it's handled in `taskrun`, but it did feel weird to me. I'll clean it up.